src/compiler/gcc.rs

```diff
@@ -208,8 +209,19 @@ where
             Some(NeedDepTarget) => need_explicit_dep_target = true,
             Some(DepTarget) => dep_target = item.arg.get_value().map(OsString::from),
             Some(PreprocessorArgument) |
-            Some(PassThrough) |
-            Some(Language) => {}
+            Some(PassThrough) => {}
+            Some(Language) => {
+                // Set extension to one matching the language set with -x
+                let lang = item.arg.get_value().map(OsString::from);
+                let lang = lang.as_ref().map(|a| a.to_string_lossy());
+                extension = match lang.as_ref().map(|a| a.as_ref()) {
```

Comment:
Maybe this wants an `ArgumentValue::as_str` or something like that? I guess you'd have to return a `Cow<str>` because of the `to_string_lossy` in there.